src/controllers/expenseController.js

```diff
@@ -0,0 +1,117 @@
+/* eslint-disable no-console */
+'use strict';
+
+const { ExpenseService } = require('../services/expenseService');
+const expenseService = new ExpenseService();
+const userService = require('../services/userService');
+
+async function createExpense(req, res) {
+  try {
+    const { userId, spentAt, title, amount, category, note } = req.body;
+
+    if (!spentAt || !title || !amount || !userId) {
+      return res.status(400).send('All fields are required');
+    }
+
+    const user = await userService.getAllUsers({ id: userId });
+
+    if (user.length === 0) {
+      return res.status(400).send('Bad Request');
+    }
+
+    const newExpense = await expenseService.createExpense(
+      userId,
+      spentAt,
+      title,
+      amount,
+      category,
+      note,
+    );
+
+    res.status(201).json(newExpense);
+  } catch (error) {
+    console.error(error);
+    res.status(500).send('Error creating expense');
+  }
+}
+
+async function getAllExpenses(req, res) {
+  try {
+    const userId = req.query.userId;
+    const category = req.query.categories;
+    const filters = {};
+
+    if (userId) {
+      filters.userId = userId;
```

Comment:
Probably not necessary, as you could directly use:
`filters.userId = req.query.userId`
Or better:
```
const { userId, categories: category } = req.query;
const expeses = await expenseService.getAllExpenses({category, userId});
``` 

Doing so, filters will have undefined values for the missing entries, and so they can later be ignored without the if testing

Reply:
You're absolutely right. Just did the pointed changes and updated de expenseServices to ensure consistency on the getAllExpenses  method so that the class is explicitly handling userId and category. Let me know if you don't think this wasn't necessary also.